### PR TITLE
fix: Keyboard command to post reaction fails if there is a system mes…

### PR DESCRIPTION
**Pull request by MeenuyD**
Fix emoji picker issue when system message is present
https://github.com/mattermost/mattermost/issues/24976
#24976

```release-note
Fix error where +:emoji: did not react to the last message if the last message was a system message
```

### CODE REVIEW

File: app/actions/remote/reactions.ts [hunk 1]
```diff
@@ -87,23 +88,28 @@ export const removeReaction = async (serverUrl: string, postId: string, emojiNam
...
-            return addReaction(serverUrl, posts[0].id, emojiName);
+        if (latestRegularUserPost) {
+            if (add) {
+                return addReaction(serverUrl, latestRegularUserPost.id, emojiName);
+            }
+            return removeReaction(serverUrl, latestRegularUserPost.id, emojiName);
         }
-        return removeReaction(serverUrl, posts[0].id, emojiName);
+
+        return { error: 'No regular user posts found in the thread.' };
```
Comment (larkox) [inline app/actions/remote/reactions.ts:110]: To make this a bit shallower you can do this instead:

```
if (!latestRegularUserPost) {
    return { error: 'No regular user posts found in the thread.' };
}

if (add) {
...
```

File: app/actions/remote/reactions.ts [hunk 2]
```diff
@@ -87,23 +88,28 @@ export const removeReaction = async (serverUrl: string, postId: string, emojiNam
         return {error};
     }
 };
-
 export const handleReactionToLatestPost = async (serverUrl: string, emojiName: string, add: boolean, rootId?: string) => {
     try {
-        const {database} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const { database } = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
```
Comment (larkox) [inline app/actions/remote/reactions.ts:93]: I expect some lint issues here. Please run `npm run fix` and most of them will be solved automatically.

File: app/actions/remote/reactions.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import {forceLogoutIfNecessary} from './session';
 
 import type {Model} from '@nozbe/watermelondb';
 import type PostModel from '@typings/database/models/servers/post';
+import { isSystemMessage } from '@app/utils/post';
```
Comment (larkox) [inline app/actions/remote/reactions.ts:18]:
```suggestion
import { isSystemMessage } from '@utils/post';
```

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks! Great work! A few minor comments.

File: app/actions/remote/reactions.ts (hunk 1 above)
Comment (enahum) [reply app/actions/remote/reactions.ts:110]: Is the error visible to the user? If so, it should be translatable

**Comment by harshilsharma63** [comment]
> Hello @MeenuyD,
> 
> Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).
> 
> Per the [Mattermost Contribution Guide](https://developers.mattermost.com/contribute/getting-started/), we need to add you to the list of [approved contributors](https://docs.google.com/spreadsheets/d/1NTCeG-iL_VS9bFqtmHSfwETo5f-8MQ7oMDE5IUYJi_Y/pubhtml?gid=0&single=true) for the Mattermost project.
> 
> **Please help complete the Mattermost [contribution license agreement](http://www.mattermost.org/mattermost-contributor-agreement/)?** Once you have signed the CLA, please comment with `/check-cla` and confirm that the CLA check is green.
> 
> This is a standard procedure for many open source projects.
> 
> Please let us know if you have any questions.
> 
> We are very happy to have you join our growing community! If you're not yet a member, please consider joining our [Contributors community channel](https://pre-release.mattermost.com/core/channels/tickets) to meet other contributors and discuss new opportunities with the core team.

@MeenuyD can you please sign the CLA?

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]
Awaiting CLA.

**Comment by MeenuyD** [comment]
Hello @harshilsharma63 what should i do for cla

**Comment by MeenuyD** [comment]
Hello @harshilsharma63 I have sign the cla

**Comment by larkox** [comment]
@MeenuyD You have a few linting issues. `npm run fix` should fix them all (or the majority). Can you take a look?

**Pull request closed without merging**